File: frontend/src/types/visit.ts
import type { VisitStatus } from "./enums";

export type { VisitStatus };

export interface Visit {
  id: string;
  appointmentId: string;
  appointmentDate: string;
  appointmentStartTime: string;
  appointmentEndTime: string;
  patientId: string;
  patientFullName: string;
  patientPhoneNumber: string;
  doctorProfileId: string;
  doctorFullName: string;
  serviceName: string;
  visitDate: string;
  status: VisitStatus;
  chiefComplaint: string | null;
  diagnosisNote: string | null;
  treatmentNote: string | null;
  toothNumbers: string | null;
  prescriptionNote: string | null;
  followUpDate: string | null;
  internalNotes: string | null;
  startedAtUtc: string | null;
  completedAtUtc: string | null;
  createdAtUtc: string;
  updatedAtUtc: string | null;
}

export interface VisitListItem {
  id: string;
  appointmentId: string;
  patientId: string;
  patientFullName: string;
  patientPhoneNumber: string;
  doctorProfileId: string;
  doctorFullName: string;
  serviceName: string;
  visitDate: string;
  status: VisitStatus;
  followUpDate: string | null;
}

export interface StartVisitRequest {
  chiefComplaint?: string;
}

export interface UpdateVisitRequest {
  chiefComplaint?: string;
  diagnosisNote?: string;
  treatmentNote?: string;
  toothNumbers?: string;
  prescriptionNote?: string;
  followUpDate?: string;
  internalNotes?: string;
}

export interface CompleteVisitRequest extends UpdateVisitRequest {}

export interface VisitQuery {
  search?: string;
  patientId?: string;
  doctorId?: string;
  status?: VisitStatus;
  fromDate?: string;
  toDate?: string;
  pageNumber?: number;
  pageSize?: number;
}

export interface VisitStats {
  totalVisits: number;
  inProgressVisits: number;
  completedVisits: number;
  followUpsScheduled: number;
}

File: backend/Entities/Visit.cs
namespace ClinicFlow.Api.Entities;

/// <summary>
/// The clinical record for a single appointment: chief complaint,
/// doctor-entered diagnosis/treatment notes, a manually typed prescription,
/// and an optional follow-up date. The doctor writes every clinical field —
/// nothing here is system-generated or AI-assisted. At most one Visit exists
/// per Appointment.
/// </summary>
public class Visit
{
    public Guid Id { get; set; }

    public Guid AppointmentId { get; set; }
    public Appointment? Appointment { get; set; }

    public Guid PatientId { get; set; }
    public Patient? Patient { get; set; }

    public Guid DoctorProfileId { get; set; }
    public DoctorProfile? DoctorProfile { get; set; }

    public DateOnly VisitDate { get; set; }
    public DateTime? StartedAtUtc { get; set; }
    public DateTime? CompletedAtUtc { get; set; }

    public VisitStatus Status { get; set; } = VisitStatus.InProgress;

    public string? ChiefComplaint { get; set; }
    public string? DiagnosisNote { get; set; }
    public string? TreatmentNote { get; set; }
    public string? ToothNumbers { get; set; }
    public string? PrescriptionNote { get; set; }
    public DateOnly? FollowUpDate { get; set; }
    public string? InternalNotes { get; set; }

    public DateTime CreatedAtUtc { get; set; }
    public DateTime? UpdatedAtUtc { get; set; }
}
